src/components/SideNav/SideNav.tsx

```diff
@@ -26,6 +26,28 @@ const SideNav = (props: SideNavProps): ReactElement => {
 		link.onClick();
 	};
 
+	const findLinkselected = () => {
+		let linkSelected: Link
+		links.forEach(link => link.forEach(l => l.label === defaultLink ? linkSelected = l : null))
+		if (!linkSelected) {
+			throw new Error("Default link is not in the links array.");
+		} else {
+			return linkSelected
+		}
+	}
+
+	useEffect(() => {
+		if  (links) {
+			if (defaultLink) {
+				setSelectedLink(defaultLink)
```

Comment:
I think this approach might break in some scenarios. For example: Here we're first assigning the default link to the selected link and then we're validating that it exists. I think we should first validate and then (if it exists) assign it.

Also... if we look at what the "onLinkClicked" function does we'll see that there we're already doing a "setSelectedLink" to whatever the parameter is, so here we're doing it twice.